Libraries/CommonFunctions.psm1

```diff
@@ -117,19 +70,36 @@ Function Select-TestCases($TestXMLs, $TestCategory, $TestArea, $TestNames, $Test
     $WildCards = @('^','.','[',']','?','+','*')
     $ExcludedTestsCount = 0
 
-    # if the expected filter parameter is 'ALL' or empty ("ALL" -imatch "" return $True), the actually filter used in this function will be '*', except for '$ExcludedTests'
-    if ("All" -imatch $TestCategory)   { $TestCategory = "*" }
-    if ("All" -imatch $TestArea)       { $TestArea = "*" }
-    if ("All" -imatch $TestNames)      { $TestNames = "*" }
-    if ("All" -imatch  $TestTag)       { $TestTag = "*" }
-    if ("All" -imatch  $TestPriority)  { $TestPriority = "*" }
-    if ("All" -imatch  $TestSetup)     { $TestSetup = "*" }
+    # if the expected filter parameter is 'All' (case insensitive) or empty or $null, the actually filter used in this function will be '*', except for '$ExcludedTests'
+    if (!$TestCategory -or ($TestCategory -eq "All"))   { $TestCategory = "*" }
+    if (!$TestArea -or ($TestArea -eq "All"))           { $TestArea = "*" }
+    if (!$TestNames -or ($TestNames -eq "All"))         { $TestNames = "*" }
+    if (!$TestTag -or ($TestTag -eq "All"))             { $TestTag = "*" }
+    if (!$TestPriority -or ($TestPriority -eq "All"))   { $TestPriority = "*" }
+    if (!$TestSetup -or ($TestSetup -eq "All"))         { $TestSetup = "*" }
 
-    $testCategoryArray = @($TestCategory.Trim(', ').Split(',').Trim())
-    $testAreaArray = @($TestArea.Trim(', ').Split(',').Trim())
-    $testNamesArray = @($TestNames.Trim(', ').Split(',').Trim())
-    $testSetupTypeArray = @($TestSetup.Trim(', ').Split(',').Trim())
-    $excludedTestsArray = @($ExcludeTests.Trim(', ').Split(',').Trim())
+    $testCategoryArray = $testAreaArray = $testNamesArray = $testTagArray = $testPriorityArray = $testSetupTypeArray = $excludedTestsArray = @()
+    if ($TestCategory -ne "*") {
+        $testCategoryArray = @($TestCategory.Trim(', ').Split(',').Trim())
```

Comment:
Can be moved to else part as mentioned in first comment.

Reply:
updated